### ISSUE TITLE
1 line if statement is wrong working in middleware with await

### ISSUE THREAD

**Issue by kyejune** [nuxt/nuxt#23550]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v18.17.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

[online-if.zip](https://github.com/nuxt/nuxt/files/12828352/online-if.zip)
>> it is a project file;

### Describe the bug

If you create an if statement in a middleware file with the wait without brace, it does not work correctly.

export default defineNuxtRouteMiddleware(async (to, from) => {
    const bool = false;

    /* CASE 1: It should not be executed, but it is */
    if( bool ) await console.log('It should not be executed, but it is');
    // else console.log('run') /* CASE 2:  SyntaxError occurs when else syntax is used */
});

### Additional context

<img width="1350" alt="image" src="https://github.com/nuxt/nuxt/assets/8554821/dd6ff8cc-d86d-4150-9d6c-a8a2e00a14f9">

**Comment by huang-julien** [comment]
Hello this is an issue with the transformation to keep the nuxt context on an `await` promise. 
Try 
```ts
  if (bool) {
    await console.log('It should not be executed, but it is');
  } else {
    console.log('run');
  }
```
instead

We should probably document this

**Comment by akintoluvic** [comment]
I would like to help with this, @huang-julien is there any guide or information to follow in resolving this.

**Comment by huang-julien** [comment]
Thanks @akintoluvic  ! We should start by adding this to the documentation first. Here's the[ contribution guide](https://nuxt.com/docs/community/contribution#writing-documentation) of nuxt :)  

I think i saw a similar issue a few months ago (i'll try to find it :eyes: ).

any thoughts @danielroe ? i'm pretty sure this is an `unctx` bug

**Comment by danielroe** [comment]
Yes, this sounds like an `unctx` bug we should fix directly in https://github.com/unjs/unctx/blob/main/src/transform.ts

**Comment by OrbisK** [comment]
I have updated the reproduction. It seems to be fixed now.
